javaslang/src/test/java/javaslang/collection/JavaConverters_ListViewMutableTest.java

```diff
@@ -0,0 +1,246 @@
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@RunWith(Parameterized.class)
+public class JavaConverters_ListViewMutableTest extends JavaConverters_ListViewBaseTest {
+
+    public JavaConverters_ListViewMutableTest(String name, ListFactory<?> listFactory) {
+        super(listFactory);
+    }
+
+    @Parameterized.Parameters(name = "{index}: {0}")
+    public static java.util.Collection<Object[]> data() {
+        // DEV-NOTE: It is sufficient to test
+        // - one standard Java impl (<-- ensures our tests are correct)
+        // - one LinearSeq impl
+        // - one IndexedSeq impl
+        // to ensure the correctness of the mutable ListView impl.
+        return java.util.Arrays.asList(new Object[][] {
+                { java.util.ArrayList.class.getName(), ListFactory.of(ts -> {
+                    final java.util.List<Object> list = new java.util.ArrayList<>();
+                    java.util.Collections.addAll(list, ts);
+                    return list;
+                }) },
+                { List.class.getName(), ListFactory.of(ts -> List.of(ts).asJavaMutable()) },
+                { Vector.class.getName(), ListFactory.of(ts -> Vector.of(ts).asJavaMutable()) }
+        });
+    }
+
+    // -- add(T)
+
+    @Test
+    public void shouldAddElementToEmptyListView() {
+        final java.util.List<Integer> list = empty();
+        assertThat(list.add(1)).isTrue();
+        assertThat(list).isEqualTo(javaList(1));
+    }
+
+    @Test
+    public void shouldAddElementToEndOfNonEmptyListView() {
+        final java.util.List<Integer> list = of(1);
+        assertThat(list.add(2)).isTrue();
+        assertThat(list).isEqualTo(javaList(1, 2));
+    }
+
+    @Test
+    public void shouldAddSubtypeToEndOfNonEmptyListView() {
+        abstract class A {
+            @Override
+            public final int hashCode() {
+                return 0;
+            }
+        }
+        final class B extends A {
+            @Override
+            public boolean equals(Object o) {
+                return o instanceof B;
+            }
+        }
+        final class C extends A {
+            @Override
+            public boolean equals(Object o) {
+                return o instanceof C;
+            }
+        }
+        final java.util.List<A> list = of(new B());
+        assertThat(list.add(new C())).isTrue();
+        assertThat(list).isEqualTo(javaList(new B(), new C()));
+    }
+
+    @Test
+    public void shouldAppendNull() {
+        final java.util.List<Object> list = empty();
+        assertThat(list.add(null)).isTrue();
+        assertThat(list).isEqualTo(javaList((Object) null));
+    }
+
+    @Test
+    public void shouldAddSelf() {
+        final java.util.List<Object> list = empty();
+        assertThat(list.add(list)).isTrue();
```

Comment:
\[FindBugs\] INFO: IL: A collection is added to itself in javaslang\.collection\.JavaConverters\_ListViewMutableTest\.shouldAddSelf\(\)